### PR TITLE
Install python packages into venvs across workflows.

### PR DESCRIPTION
**Pull request by ScottTodd**
Many of these workflows are using persistent self-hosted runners, so it looks like they have been reusing the same system-wide Python environment between workflow runs (plus layer of caching on top). This switches to using venvs at `${{ github.workspace }}/.venv` that should be ephemeral, giving us more explicit control over which packages are installed.

More work is planned as part of https://github.com/nod-ai/shark-ai/issues/584 to refactor these workflows further - replacing the package installs code like `pip install --no-compile -r requirements.txt -r sharktank/requirements-tests.txt -e sharktank/` with a `setup_venv.py` script that uses dev/nightly/stable packages (from an appropriate source).

This also disables pip caching, since that is not directly compatible with using venvs. As a result, some workflows are slower now, but they are more predictable in what they install. Good reading for adding caching back:
* https://adamj.eu/tech/2023/11/02/github-actions-faster-python-virtual-environments/
* https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#caching-packages

### CODE REVIEW

File: .github/workflows/ci-sharktank.yml [hunk 1]
```diff
@@ -98,7 +97,6 @@ jobs:
```
Comment (archana-ramalingam) [inline .github/workflows/ci-sharktank.yml:97]: We could add missed changes from [here](https://github.com/nod-ai/shark-ai/pull/639#discussion_r1868485174).

Comment (ScottTodd) [reply .github/workflows/ci-sharktank.yml:97]: Keeping PRs small and focused. Sent https://github.com/nod-ai/shark-ai/pull/642 for that.

File: .github/workflows/ci-llama-quick-tests.yaml [hunk 2]
```diff
@@ -47,6 +47,8 @@ jobs:
         uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
         with:
           python-version: ${{matrix.version}}
+      - name: Create Python venv
+        run: python -m venv ${VENV_DIR}
 
       - name: Cache Pip Packages
```
Comment (ScottTodd) [inline .github/workflows/ci-llama-quick-tests.yaml:53]: Notes to self:

Need to be deliberate with pip caching. I think I want the downloads to be cached, but _not_ any installed packages.

Reference https://adamj.eu/tech/2023/11/02/github-actions-faster-python-virtual-environments/ and https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#caching-packages.

Both of those approaches hash the requirements files as the cache key, but we have so many separate requirements files _and_ we install our own dev and nightly packages on top, which are not pinned in any requirement files to be cached.

Some of that will hopefully be less consequential, if/when we switch to the `setup_venv.py` + `uv` setup style that I'm prototyping on https://github.com/nod-ai/shark-ai/pull/625.

Comment (ScottTodd) [reply .github/workflows/ci-llama-quick-tests.yaml:53]: Maybe try landing tomorrow without any pip/venv caching, then figure out how to add it back carefully.  Some extra CI time is worth the environment consistency.

File: .github/workflows/ci-sglang-benchmark.yml [hunk 3]
```diff
@@ -46,6 +46,7 @@ jobs:
         shell: bash
     env:
       PIP_CACHE_DIR: "${{ github.workspace }}/.pip-cache"
```
Comment (stbaione) [inline .github/workflows/ci-sglang-benchmark.yml:48]: `PIP_CACHE_DIR` can be removed

File: .github/workflows/ci-sglang-integration-tests.yml [hunk 4]
```diff
@@ -35,6 +35,7 @@ jobs:
         shell: bash
     env:
       PIP_CACHE_DIR: "${{ github.workspace }}/.pip-cache"
```
Comment (stbaione) [inline .github/workflows/ci-sglang-integration-tests.yml:37]: `PIP_CACHE_DIR` can be removed

File: .github/workflows/ci-sglang-benchmark.yml (hunk 3 above)
Comment (ScottTodd) [reply .github/workflows/ci-sglang-benchmark.yml]: Ah, thanks. Removed.

**Review by stbaione — APPROVED** [review]

**Review by marbre — APPROVED** [review]

**Pull request merged**